feat(#692): Challenge 34 with a focus on determenistic use of KDFs (#866)
OWASP/wrongsecrets: src/main/resources/explanations/challenge34_reason.adoc

> *Why Key Derivation Functions are not safe when using hardcoded values*
> 
> Key Derivation Functions (KDFs) are deterministic in nature. This means that for a given input, they will always give back the same output. So, if the parameters are hardcoded in code, anyone with access to the code can run the KDF with the coded parameters and get the actual key.
> 
> KDFs should be used to generate keys based on dynamic input, such as human supplied passwords for instance. KDFs might be used in situations where secure random sources generate certain output which can be used to generate keys. Again, in both cases: the KDF its input is not hardcoded/deterministic.

Reviewer: ```suggestion
*Why Key Derivation Functions are not safe when using hardcoded values*

Key Derivation Functions (KDFs) are deterministic. This means that they will always give back the same output for a given input. So, if the parameters are hardcoded, anyone with access to the code can run the KDF with the specified parameters and get the key.

KDFs should be used to generate keys based on dynamic input, such as human-supplied passwords. KDFs might be used when secure random sources can be used to generate keys. Again, the KDF's input is not hardcoded/deterministic in both cases.
```
